airflow/docker.go

```diff
@@ -589,6 +590,32 @@ func Deploy(path, name, wsId string, prompt bool) error {
 	// Build our image
 	fmt.Println(messages.COMPOSE_IMAGE_BUILDING_PROMT)
 
+	// parse dockerfile
+	cmds, err := docker.ParseFile(filepath.Join(path, "Dockerfile"))
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse dockerfile: %s", filepath.Join(path, "Dockerfile"))
+	}
+
+	image, tag := docker.GetImageTagFromParsedFile(cmds)
+	if image != messages.VALID_DOCKERFILE_BASE_IMAGE {
+		fmt.Printf("WARNING!!! You are using invalid image name '%s' in your Dockerfile, please replace to %s\n", image, messages.VALID_DOCKERFILE_BASE_IMAGE)
```

Comment:
This is a little bit heavy handed of a message. I can imagine some of our users would create a new base image for thier company that have libraries/utils pre-installed in to the image, so we need a way of silencing this warning -- probably by something in the .astro project file?

Reply:
may be rephrase message to:
`You are using custom image name... on your own risk? `

Reply:
Hmm, it may come down to how we plan to deal with the airflow version on the server side. If a user wants to silence this warning, they'd also want to turn off any (future) guardrails that we have in place in Houston. 

Was chatting with @kaxil about this earlier today some. Feels like we're on the completely "loose" end of the spectrum now, but we could build up some guardrails that key on our image labels, or we can go all out and completely manage the image ourselves and give the users less control, but that scares me a little bit.

I like the guardrails approach we've discussed in the past, but just trying to connect the dots on how this would tie in.

Reply:
My 5 cents:
Right now we have "Certified" airflow image and it's good to WARN for user if it uses something different on its own risk. 

To skip warning user can change his custom image and reuse authorized `-onbuild` one or propose pull request to add new image to `ap-airflow` repo.

Reply:
I'm thinking of this sort of use case:

```
FROM astronomerinc/ap-airflow:1.10.5-1

RUN pip install a-big-slow-module
```

That is then `docker build -t my-co/ap-airflow-base`, and then the docker file for the astro project looks like

```
FROM my-co/ap-airflow-base
```

I think that should be supported, and _if configured_ not print any warnings. A pull request to `ap-airflow` isn't appropriate for this sort of "baseimage" as it's specific to one company, not all of our users.

Reply:
@ashb 
yeah, to cover this case i need to docker pull/read image meta info, which is overkill. Probably if we can introduce astro flag/ or config it would be ok for now.

Reply:
Yeah, I agree it would be overkill. I think just having this warning disablely by a setting in the .astro/config.yml file would be the way to go.

(Also we don't have a confirm step here, just on the tag one? Was that on purpose?)

Reply:
@ash something like it?
```
cat ~/.astro/config.yaml
...
show_warnings: true
```
?

Reply:
Yep, adding a config to suppress the warning to the project makes sense to me.

Reply:
done.